### PR TITLE
wallet: allow message sign/verify for subaddresses

### PR DESCRIPTION
**Pull request by moneromooo-monero**

### CODE REVIEW

File: src/simplewallet/simplewallet.cpp [hunk 1]
```diff
@@ -214,7 +214,7 @@ namespace
   const char* USAGE_GET_TX_NOTE("get_tx_note <txid>");
   const char* USAGE_GET_DESCRIPTION("get_description");
   const char* USAGE_SET_DESCRIPTION("set_description [free text note]");
-  const char* USAGE_SIGN("sign <filename>");
+  const char* USAGE_SIGN("sign [<account_index>,<address_index>] <filename>");
```
Comment (vtnerd) [inline src/simplewallet/simplewallet.cpp]: Why move the filename position? Putting the indexes after would mean existing usage remains unchanged (defaults to primary account).

Comment (moneromooo-monero) [reply src/simplewallet/simplewallet.cpp]: Because it's what other commands do. AFAIK existing usage remains unchanged. Did you try it and it changed ?

Comment (vtnerd) [reply src/simplewallet/simplewallet.cpp]: No, I stunk at reading the code last time - too quickly again.. Let me try again.

File: src/simplewallet/simplewallet.cpp [hunk 2]
```diff
@@ -9609,7 +9609,20 @@ bool simple_wallet::sign(const std::vector<std::string> &args)
...
+    if (sscanf(args[0].c_str(), "%u,%u", &a, &b) != 2)
+    {
+      fail_msg_writer() << tr("Invalid subaddress index format");
+      return true;
+    }
+    index.major = a;
+    index.minor = b;
+  }
+
+  std::string filename = args.back();
```
Comment (vtnerd) [inline src/simplewallet/simplewallet.cpp]: A copy doesn't appear to be needed here -> `const std::string&` ?

File: src/wallet/wallet2.cpp [hunk 3]
```diff
@@ -11777,13 +11777,27 @@ void wallet2::set_account_tag_description(const std::string& tag, const std::str
...
+  crypto::public_key pkey;
+  if (index.is_zero())
+  {
+    skey = keys.m_spend_secret_key;
+    pkey = keys.m_account_address.m_spend_public_key;
+  }
+  else
+  {
+    skey = keys.m_spend_secret_key;
+    crypto::secret_key m = m_account.get_device().get_subaddress_secret_key(m_account.get_keys().m_view_secret_key, index);
```
Comment (vtnerd) [inline src/wallet/wallet2.cpp]: `m_account.get_keys()` -> `keys` ... the alias was used elsewhere already.
